server/server_test.go

```diff
@@ -386,3 +386,34 @@ func TestNilMonitoringPort(t *testing.T) {
 		t.Fatal("HttpAddr should be nil.")
 	}
 }
+
+type DummyAuth struct {
+	count int
+}
+
+func (d *DummyAuth) Check(c ClientAuth) bool {
+	d.count++
+	return true
+}
+
+func TestCustomClientAuth(t *testing.T) {
+	var clientAuth DummyAuth
+
+	opts := DefaultOptions()
+	opts.CustomClientAuth = &clientAuth
+
+	s := RunServer(opts)
+
+	defer s.Shutdown()
+
+	addr := fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port)
+	nc, err := nats.Connect(addr)
+	if err != nil {
+		t.Fatalf("Error creating client: %v\n", err)
+	}
+	nc.Close()
+
+	if clientAuth.count != 1 {
+		t.Error("Client auth should have been delegated to CustomClientAuth")
+	}
```

Comment:
With changes proposed above, that would simplify and check that `Connect()` actually fails. Here is a modified version along with Router check:
```
func TestCustomClientAuth(t *testing.T) {
	opts := DefaultOptions()
	opts.CustomClientAuth = &DummyAuth{}

	s := RunServer(opts)
	defer s.Shutdown()

	addr := fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port)
	if _, err := nats.Connect(addr); err == nil {
		t.Fatal("Expected client to fail to connect")
	}
}

func TestCustomRouterAuth(t *testing.T) {
	opts := DefaultOptions()
	opts.CustomRouterAuth = &DummyAuth{}
	opts.Cluster.Host = "127.0.0.1"
	s := RunServer(opts)
	defer s.Shutdown()
	clusterPort := s.ClusterAddr().Port

	opts2 := DefaultOptions()
	opts2.Cluster.Host = "127.0.0.1"
	opts2.Routes = RoutesFromStr(fmt.Sprintf("nats://127.0.0.1:%d", clusterPort))
	s2 := RunServer(opts2)
	defer s2.Shutdown()
	if nr := s2.NumRoutes(); nr != 0 {
		t.Fatalf("Expected no route, got %v", nr)
	}
}
```